ui/packages/tidb-dashboard-lib/src/components/MetricChart/index.tsx

```diff
@@ -179,6 +184,9 @@ export default function MetricChart({
           }
         }
         fillInto[fillIdx] = data
+        if (data?.result.length === 0) {
+          setHasEmptyQueryResult(true)
```

Comment:
It would be better to set "empty state" in a single batch process. How about do this in `queryAllMetrics` like:
```
const emptyMetrics: string[] = []
dataSets.forEach((data) => {
  ...
  if (data.result.length === 0) {
    emptyMetrics.push(format(queries[queryIdx].name, promResult.metric))
  }
  ...
})

So that we can re-render charts at once and show all empty metric names in the tooltip.
```

Reply:
Oh, react won't re-render if states have not changed, but it's still better to set state at once.

Reply:
Awesome! Thanks for your suggestions. However, we will change the interaction to give users tips when the panel gets empty result.